proto/storagepb.proto

```diff
@@ -0,0 +1,136 @@
+syntax = "proto3";
+package storagepb;
+
+import "gogoproto/gogo.proto";
+import "rustproto.proto";
+
+option (gogoproto.sizer_all) = true;
+option (gogoproto.marshaler_all) = true;
+option (gogoproto.unmarshaler_all) = true;
+option (rustproto.lite_runtime_all) = true;
+
+option java_package = "org.tikv.kvproto";
+
+// Storage is the storage service for PD.
+service Storage {
+    rpc Watch(WatchRequest) returns (stream WatchResponse) {}
+
+    rpc Get(GetRequest) returns (GetResponse) {}
+
+    rpc Put(PutRequest) returns (PutResponse) {}
+}
+
+enum ErrorType {
+    OK = 0;
+    UNKNOWN = 1;
+    NOT_BOOTSTRAPPED = 2;
+    ALREADY_BOOTSTRAPPED = 3;
+    INCOMPATIBLE_VERSION = 4;
+    INVALID_VALUE = 5;
+    // required watch revision is smaller than current compact/min revision.
+    DATA_COMPACTED = 6;
+}
+
+message Error {
+    ErrorType type = 1;
+    string message = 2;
+}
+
+message RequestHeader {
+    // cluster_id is the ID of the cluster which be sent to.
+    uint64 cluster_id = 1;
+    // source is the source of the request.
+    string source = 2;
+}
+
+message ResponseHeader {
+    // cluster_id is the ID of the cluster which sent the response.
+    uint64 cluster_id = 1;
+    Error error = 2;
+    int64 revision = 3;
+}
+
+// copied part of https://github.com/etcd-io/etcd/blob/7dfd29b0cc7ce25337276dce646ca2a65aa44b4d/api/etcdserverpb/rpc.proto
+message WatchRequest {
+    bytes key = 1;
+    bytes range_end = 2;
+    int64 start_revision = 3;
+}
+
+// copied part of https://github.com/etcd-io/etcd/blob/7dfd29b0cc7ce25337276dce646ca2a65aa44b4d/api/etcdserverpb/rpc.proto
+message WatchResponse {
+    ResponseHeader header = 1;
+    int64 compact_revision = 2;
+    repeated Event events = 3;
+}
+
+// copied part of https://github.com/etcd-io/etcd/blob/7dfd29b0cc7ce25337276dce646ca2a65aa44b4d/api/etcdserverpb/rpc.proto
+message GetRequest {
```

Comment:
Do you think we need to add more comments to indicate that the `GetRequest` here corresponds to etcd's `RangeRequest`?

Reply:
Both are ok if we are going to use another kind of storage, Get sounds more general.

Reply:
add a comment in RPC definition